### DISCUSSION TITLE
Can we add async api?

### DISCUSSION THREAD

**Discussion by wy96f** [langgenius/dify#11257] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

The workflow api may last a few minutes. If there are many concurrent requests, this will lead to a surge in client threads and network connections, affecting the stability of the system.

### 2. Additional context or comments

Can we add async api?
For example, asyn api returns taskId directly and Client can query task by taskId.

### 3. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by huangyafei** [comment]
I think this feature is very practical.

Currently, I am using a very inconvenient way to implement asynchronous API.

I randomly generate a user ID, then request the workflow API, then obtain the workflow run ID corresponding to the user ID through the log endpoint, and finally obtain the final workflow result through the workflow run ID.

But I think this method is not elegant.

**Comment by huynhbaobk** [comment]
Any plan for this feature?

**Comment by ipcpu** [comment]
I need this feature. I just need to trigger the workflow, and the execution results will be sent via email within the workflow.

Moreover, I have many workflows that need to be triggered simultaneously, and I don't want to wait for the api results.

**Comment by kinoooolu** [comment]
Any plan to implement this feature?

**Comment by shesung** [comment]
any progress？

**Comment by KylinAnn** [comment]
any progress? we doooo need this!

**Comment by fniu** [comment]
We really need this feature! Thanks.
